app/controllers/reservations_controller.rb

```diff
@@ -0,0 +1,40 @@
+class ReservationsController < ApplicationController
+  before_action :set_reservation, only: %i[show destroy]
+  before_action :authorized
+
+  def index
+    @reservation = Reservation.all
+
+    render json: @reservation
+  end
+
+  def show
+    render json: @reservation
+  end
+
+  def create
+    @reservation = Reservation.new(reservation_params)
+    @reservation.user = @user
+
+    if @reservation.save
+      render json: @reservation, status: :created, location: @reservation
+    else
+      render json: @reservation.errors, status: :unprocessable_entity
+    end
+  end
+
+  def destroy
+    @reservation.destroy
+  end
```

Comment:
- [ ] What happens here if the reservation is not successfully destroyed? Kindly, add some error handling, like you did on the `create` method (line 15). 👍